ormar/models/newbasemodel.py

```diff
@@ -213,6 +223,26 @@ def __setstate__(self, state: Dict[Any, Any]) -> None:
         for name, value in relations.items():
             setattr(self, name, value)
 
+    def _update_relation_cache(self, prev_hash: int, new_hash: int) -> None:
```

Comment:
For review: There might be a better way to do this. Basically, I just wanted to make sure to update the cache when the current model changes--this code block basically makes sure it either updates its own cache, or updates the cache of its related model when it's in the related model list. 

It's slightly inefficient since it doesn't actually check that the foreign relation list is of the same type as the current model--but I wasn't sure how to do that. The list check is O(1) time so I didn't think it would be too big of an issue.